common/libzkp/impl/src/batch.rs

```diff
@@ -126,12 +143,13 @@ pub unsafe extern "C" fn gen_batch_proof(
         check_chunk_hashes("", &chunk_hashes_proofs).map_err(|e| format!("failed to check chunk info: {e:?}"))?;
 
         let batch = BatchProvingTask {
-            chunk_proofs
+            chunk_proofs,
+            batch_header,
```

Comment:
this `gen_batch_proof` logic is valid on darwin only, not suitable for curie.
actually, since the libzkp is not used in `generating` any proofs, we can remove all the `gen_xxx_proof` function in it